### ISSUE TITLE
[tests] fadvise is failing on a big-endian system

### ISSUE THREAD

**Issue by mator** [util-linux/util-linux#2249]
Running fadvise test on a BE system (gcc203, ppc64) , returns error.
```
gcc203:~/util-linux.git/tests$ ./run.sh fadvise

-------------------- util-linux regression tests --------------------

                    For development purpose only.                    
                 Don't execute on production system!                 

       kernel: 6.0.0-6-powerpc64                 

      options: --srcdir=/home/mator/util-linux.git/tests/.. \
               --builddir=/home/mator/util-linux.git/tests/..

      fadvise: drop page caches related to a file ... FAILED (fadvise/drop)

---------------------------------------------------------------------
  1 tests of 1 FAILED

      fadvise/drop
---------------------------------------------------------------------

$ ../test_sysinfo 
       WORDSIZE: 64
       pagesize: 65536
        INT_MAX: 2147483647
       UINT_MAX: 4294967295
       LONG_MAX: 9223372036854775807
      ULONG_MAX: 18446744073709551615
    ULONG_MAX32: 4294967295
     UINT64_MAX: 18446744073709551615
     byte-order: BE
      wcsspn-ok: 1
      enotty-ok: 1
      fsopen-ok: 1
```
running with diff:
```
util-linux.git/tests$ ./run.sh --show-diff fadvise

-------------------- util-linux regression tests --------------------

                    For development purpose only.                    
                 Don't execute on production system!                 

       kernel: 6.0.0-6-powerpc64                 

      options: --show-diff \
               --srcdir=/home/mator/util-linux.git/tests/.. \
               --builddir=/home/mator/util-linux.git/tests/..

      fadvise: drop page caches related to a file ...
diff-{{{
--- /home/mator/util-linux.git/tests/expected/fadvise/drop      2022-10-12 16:37:42.000000000 +0300
+++ /home/mator/util-linux.git/tests/output/fadvise/drop        2023-05-21 15:03:15.963413421 +0300
@@ -1,5 +1,5 @@
-  RES PAGES SIZE FILE
-  32K     8  32K ddtest
+RES PAGES SIZE FILE
+ 0B     0  32K ddtest
 
 whole file
 status: 0
@@ -9,15 +9,15 @@
 offset: 8192
 status: 0
 RES PAGES SIZE FILE
- 8K     2  32K ddtest
+ 0B     0  32K ddtest
 
 length: 16384
 status: 0
-  RES PAGES SIZE FILE
-  16K     4  32K ddtest
+RES PAGES SIZE FILE
+ 0B     0  32K ddtest
 
 offset: 8192, length: 16384 fd: 42
 status: 0
-  RES PAGES SIZE FILE
-  16K     4  32K ddtest
+RES PAGES SIZE FILE
+ 0B     0  32K ddtest
 
}}}-diff

 FAILED (fadvise/drop)

---------------------------------------------------------------------
  1 tests of 1 FAILED

      fadvise/drop
---------------------------------------------------------------------
```

problems:
1. after initial dd, test file `ddfile` does not hit kernel page cache, so `fadvise` returns always 0.
2. and if i `cat ddfile` it returns different numbers on a big endian system, like:
```
tests$ cat ddtest
tests$ ../fincore ddtest
  RES PAGES SIZE FILE
  64K     1  32K ddtest
```

**Comment by t-8ch** [comment]
For 1: Which filesystem is this on?
For 2: I guess the problem is less with big endian, but the 64k pagesize. The tests seem to assume 4K page sizes.

**Comment by t-8ch** [comment]
Note: I can reproduce both issues.
Issue 1) seems also be due to the bigger page size.
If the page is not fully written then it won't end up in the pagecache.

**Comment by t-8ch** [comment]
Should be fixed with #2251
